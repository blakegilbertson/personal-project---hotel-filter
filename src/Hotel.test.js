import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import HotelList from './Hotel';

configure({ adapter: new Adapter() });

describe('<HotelList /> shallow renders correctly', () => {
  const hotelList = shallow(<HotelList />);
  it('matches snapshot', () => {
    expect(toJson(hotelList)).toMatchSnapshot();
  });
  it('h1 should contain correct heading text', () => {
    expect(hotelList.find('h1').text()).toBe('Hotel Filtering');
  });
  it('Table outputs with headings and some data', () => {
    expect(hotelList.find('.hotel-table').exists()).toBeTruthy();
    expect(hotelList.find('.hotel-table th').exists()).toBeTruthy();
    expect(hotelList.find('.hotel-table td').exists()).toBeTruthy();
    expect(hotelList.find('.hotel-row').exists()).toBeTruthy();
  });
});
