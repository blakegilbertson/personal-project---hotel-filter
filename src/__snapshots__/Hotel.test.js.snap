// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HotelList /> shallow renders correctly matches snapshot 1`] = `
<div
  className="hotel-list"
>
  <header>
    <h1>
      Hotel Filtering
    </h1>
  </header>
  <section
    className="hotel-content"
  />
</div>
`;
